Rewrite some sections of the documentation to match what we deliver on
Solaris, or remove sections that are of no relevance to Solaris.

--- mercurial-2.7/hgext/hgk.py	Mon Aug 19 14:57:23 2013
+++ mercurial-2.7/hgext/hgk.py	Mon Aug 19 15:04:11 2013
@@ -13,13 +13,12 @@
 
 hgk consists of two parts: a Tcl script that does the displaying and
 querying of information, and an extension to Mercurial named hgk.py,
-which provides hooks for hgk to get information. hgk can be found in
-the contrib directory, and the extension is shipped in the hgext
-repository, and needs to be enabled.
+which provides hooks for hgk to get information.
 
-The :hg:`view` command will launch the hgk Tcl script. For this command
-to work, hgk must be in your search path. Alternately, you can specify
-the path to hgk in your configuration file::
+The :hg:`view` command will launch the hgk Tcl script. The script is
+shipped in /usr/demo/mercurial, and hgk needs no configuration to find
+it.  You can specify the path to an alternate hgk in your configuration
+file::
 
   [hgk]
   path=/location/of/hgk
--- mercurial-2.7/mercurial/help/hg.1.txt	Thu Aug  1 20:37:39 2013
+++ mercurial-2.7/mercurial/help/hg.1.txt	Mon Aug 19 15:14:00 2013
@@ -85,6 +85,11 @@
     repositories. The file format is the same as for ``.hgtags``, but it is
     encoded using the local system encoding.
 
+``/usr/demo/mercurial``
+    This directory contains assorted files which are part of the Mercurial
+    distribution, but not core to its functionality.  They will generally
+    need to be copied elsewhere to be of use.
+
 Some commands (e.g. revert) produce backup files ending in ``.orig``,
 if the ``.orig`` file already exists and is not tracked by Mercurial,
 it will be overwritten.
--- mercurial-3.2.3/mercurial/help/config.txt	Thu Dec 18 12:59:22 2014
+++ mercurial-3.2.3/mercurial/help/config.txt	Thu Dec 18 18:19:37 2014
@@ -18,8 +18,7 @@
 Mercurial reads configuration data from several files, if they exist.
 These files do not exist by default and you will have to create the
 appropriate configuration files yourself: global configuration like
-the username setting is typically put into
-``%USERPROFILE%\mercurial.ini`` or ``$HOME/.hgrc`` and local
+the username setting is typically put into ``$HOME/.hgrc`` and local
 configuration is put into the per-repository ``<repo>/.hg/hgrc`` file.
 
 The names of these files depend on the system on which Mercurial is
@@ -30,7 +29,7 @@
 
 .. container:: verbose.unix
 
-  On Unix, the following files are consulted:
+  The following files are consulted:
 
   - ``<repo>/.hg/hgrc`` (per-repository)
   - ``$HOME/.hgrc`` (per-user)
@@ -39,49 +38,19 @@
   - ``/etc/mercurial/hgrc`` (per-system)
   - ``/etc/mercurial/hgrc.d/*.rc`` (per-system)
 
-.. container:: verbose.windows
+Per-repository configuration options only apply in a particular
+repository.  This file is not version-controlled, and will not get
+transferred during a "clone" operation. Options in this file override
+options in all other configuration files.  Most of this file will be
+ignored if it doesn't belong to a trusted user or to a trusted group.
+See the documentation for the ``[trusted]`` section below for more
+details.
 
-  On Windows, the following files are consulted:
+Per-user configuration file(s) are for the user running Mercurial.
+Options in these files apply to all Mercurial commands executed by
+this user in any directory. Options in these files override per-system
+and per-installation options.
 
-  - ``<repo>/.hg/hgrc`` (per-repository)
-  - ``%USERPROFILE%\.hgrc`` (per-user)
-  - ``%USERPROFILE%\Mercurial.ini`` (per-user)
-  - ``%HOME%\.hgrc`` (per-user)
-  - ``%HOME%\Mercurial.ini`` (per-user)
-  - ``<install-dir>\Mercurial.ini`` (per-installation)
-  - ``<install-dir>\hgrc.d\*.rc`` (per-installation)
-  - ``HKEY_LOCAL_MACHINE\SOFTWARE\Mercurial`` (per-installation)
-
-  .. note::
-
-   The registry key ``HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Mercurial``
-   is used when running 32-bit Python on 64-bit Windows.
-
-.. container:: verbose.plan9
-
-  On Plan9, the following files are consulted:
-
-  - ``<repo>/.hg/hgrc`` (per-repository)
-  - ``$home/lib/hgrc`` (per-user)
-  - ``<install-root>/lib/mercurial/hgrc`` (per-installation)
-  - ``<install-root>/lib/mercurial/hgrc.d/*.rc`` (per-installation)
-  - ``/lib/mercurial/hgrc`` (per-system)
-  - ``/lib/mercurial/hgrc.d/*.rc`` (per-system)
-
-Per-repository configuration options only apply in a
-particular repository. This file is not version-controlled, and
-will not get transferred during a "clone" operation. Options in
-this file override options in all other configuration files. On
-Plan 9 and Unix, most of this file will be ignored if it doesn't
-belong to a trusted user or to a trusted group. See the documentation
-for the ``[trusted]`` section below for more details.
-
-Per-user configuration file(s) are for the user running Mercurial. On
-Windows 9x, ``%HOME%`` is replaced by ``%APPDATA%``. Options in these
-files apply to all Mercurial commands executed by this user in any
-directory. Options in these files override per-system and per-installation
-options.
-
 Per-installation configuration files are searched for in the
 directory where Mercurial is installed. ``<install-root>`` is the
 parent directory of the **hg** executable (or symlink) being run. For
@@ -490,12 +459,6 @@
 of an empty temporary file, where the filtered data must be written by
 the command.
 
-.. note::
-
-   The tempfile mechanism is recommended for Windows systems,
-   where the standard shell I/O redirection operators often have
-   strange effects and may corrupt the contents of your files.
-
 This filter mechanism is used internally by the ``eol`` extension to
 translate line ending characters between Windows (CRLF) and Unix (LF)
 format. We suggest you use the ``eol`` extension for convenience.
@@ -849,13 +812,6 @@
    Also, hooks like "commit" will be called in all contexts that
    generate a commit (e.g. tag) and not just the commit command.
 
-.. note::
-
-   Environment variables with empty values may not be passed to
-   hooks on platforms such as Windows. As an example, ``$HG_PARENT2``
-   will have an empty value under Unix-like platforms for non-merge
-   changesets, while it will not be available at all under Windows.
-
 The syntax for Python hooks is as follows::
 
   hookname = python:modulename.submodule.callable
@@ -951,7 +907,6 @@
 
   # Define new tool
   myHtmlTool.args = -m $local $other $base $output
-  myHtmlTool.regkey = Software\FooSoftware\HtmlMerge
   myHtmlTool.priority = 1
 
 Supported arguments:
@@ -961,9 +916,7 @@
   Default: 0.
 
 ``executable``
-  Either just the name of the executable or its pathname.  On Windows,
-  the path can use environment variables with ${ProgramFiles} syntax.
-  Default: the tool name.
+  Either just the name of the executable or its pathname.  Default: the tool name.
 
 ``args``
   The arguments to pass to the tool executable. You can refer to the
@@ -1011,29 +964,6 @@
 ``gui``
   This tool requires a graphical interface to run. Default: False
 
-``regkey``
-  Windows registry key which describes install location of this
-  tool. Mercurial will search for this key first under
-  ``HKEY_CURRENT_USER`` and then under ``HKEY_LOCAL_MACHINE``.
-  Default: None
-
-``regkeyalt``
-  An alternate Windows registry key to try if the first key is not
-  found.  The alternate key uses the same ``regname`` and ``regappend``
-  semantics of the primary key.  The most common use for this key
-  is to search for 32bit applications on 64bit operating systems.
-  Default: None
-
-``regname``
-  Name of value to read from specified registry key. Defaults to the
-  unnamed (default) value.
-
-``regappend``
-  String to append to the value read from the registry, typically
-  the executable name of the tool.
-  Default: None
-
-
 ``patch``
 ---------
 
